### ISSUE TITLE
`needless_lifetimes` ignores lifetime use in closure

### ISSUE THREAD

**Issue by TimNN** [rust-lang/rust-clippy#14607]
### Summary

If I write a statement like `let _: &'s u8 = x;`, clippy considers that a use of `'s`, and thus does not consider `'s` "needless".

However, if I write that same statement in a closure instead, clippy considers `'s` to be unused and suggests removing it.

(Actually removing `'s` then causes a compiler error).

### Lint Name

needless_lifetimes

### Reproducer

I tried this code:

```rust
#![allow(warnings)]
#![deny(clippy::needless_lifetimes)]

pub fn check<'s>(x: &'s u8) {
    // This silences the lint:
    // let _: &'s u8 = x;
    
    (||{
        let _: &'s u8 = x;
    })();
}
```

I saw this happen:

```
error: the following explicit lifetimes could be elided: 's
 --> src/lib.rs:4:14
  |
4 | pub fn check<'s>(x: &'s u8) {
  |              ^^      ^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
note: the lint level is defined here
 --> src/lib.rs:2:9
  |
2 | #![deny(clippy::needless_lifetimes)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
help: elide the lifetimes
  |
4 - pub fn check<'s>(x: &'s u8) {
4 + pub fn check(x: &u8) {
  |
```

I expected to see this happen: The lint does not trigger (as is the case if you uncomment the statement outside the closure).

### Version

```text
Latest playground nightly: (2025-04-07 e643f59f6da3a84f43e7)
```
